sklearn/linear_model/_least_angle.py

```diff
@@ -859,17 +859,19 @@ class Lars(MultiOutputMixin, RegressorMixin, LinearModel):
     verbose : bool or int, default=False
         Sets the verbosity amount.
 
-    normalize : bool, default=True
+    normalize : bool, default=False
         This parameter is ignored when ``fit_intercept`` is set to False.
         If True, the regressors X will be normalized before regression by
         subtracting the mean and dividing by the l2-norm.
         If you wish to standardize, please use
         :class:`~sklearn.preprocessing.StandardScaler` before calling ``fit``
         on an estimator with ``normalize=False``.
 
-        .. deprecated:: 1.0
-            ``normalize`` was deprecated in version 1.0. It will default
-            to False in 1.2 and be removed in 1.4.
+        .. versionchanged:: 1.2
+           default changed from True to False in 1.2.
+
+        .. deprecated:: 1.2
+            ``normalize`` was deprecated in version 1.2 and will be removed in 1.4.
```

Comment:
For my education: Is this (change the version in which it was deprecated) the normal thing to do? I was expecting to see the old comment left as is and a new one added to announce the change of the default value.

Reply:
We've never been very consistent with these directives. I found it more explicit to split them, because it what was actually done, a change of default followed by a deprecation

Reply:
I like the split as well. I think the thing I am looking for is a way to let the user know "this change has been coming since 1.0", like the current directive does. In the version from this PR it sounds like the changes started in 1.2. Thinking about it today, I am not so sure anymore if preserving this historical information is actually useful or if the version of this PR is better because it contains all the information the reader needs _now_ and not lots of additional stuff (that isn't so actionable).

You decide. I'm fine with either.

Reply:
For changes of default values, we've decided to use ``versionchanged`` targeting the version of the actual change (here 1.2) right away when initiating the change (here 1.0). I think it was in https://github.com/scikit-learn/scikit-learn/pull/23244

The information of when it was decided to initiate the change is in the changelog (here v1.0.rst), which I think is good enough.